docs(buttons): creation of buttons docs (#1879)
pankod/refine: documentation/docs/ui-frameworks/mui/components/buttons/edit.md

---
id: edit-button
title: Edit
---

import editButton from '@site/static/img/guides-and-concepts/components/buttons/edit/edit-mui.png';

> `<EditButton>` uses Material UI [`<Button>`](https://mui.com/material-ui/react-button/) component. It uses the `create` method from [`useNavigation`](/core/hooks/navigation/useNavigation.md) under the hood. It can be useful to redirect the app to the create page route of resource.

Reviewer: ```suggestion
`<EditButton>` uses Material UI [`<Button>`](https://mui.com/material-ui/react-button/) component. It uses the `edit` method from [`useNavigation`](/core/hooks/navigation/useNavigation.md) under the hood. It can be useful to redirect the app to the edit page route of resource.
```

                    </TableRow>
                </TableHead>
                <TableBody>
                    {data?.data.map((row) => (
                        <TableRow key={row.id}>
                            <TableCell>{row.id}</TableCell>
                            <TableCell component="th" scope="row">
                                {row.title}
                            </TableCell>
                            <TableCell align="center">
                                // highlight-next-line
                                <EditButton recordItemId={row.id} />
                            </TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </List>
    );
};

interface IPost {
    id: string;
    title: string;
}
```

Will look like this:

<div class="img-container">
    <div class="window">
        <div class="control red"></div>
        <div class="control orange"></div>
        <div class="control green"></div>
    </div>
    <img src={editButton} alt="Default edit button" />
</div>

## Properties

### `resourceNameOrRouteName`

> It is used to redirect the app to the `/create` endpoint of the given resource name. By default, the app redirects to a URL with `/create` defined by the name property of resource object.

Reviewer: ```suggestion
It is used to redirect the app to the `/edit` endpoint of the given resource name. By default, the app redirects to a URL with `/edit` defined by the name property of the resource object.
```

```tsx
import { EditButton } from "@pankod/refine-mui";

export const MyCreateComponent = () => {

>     return <EditButton resourceNameOrRouteName="posts" />;

Reviewer: ```suggestion
    return <EditButton resourceNameOrRouteName="posts" recordItemId="2" />;
```

};
```

> Clicking the button will trigger the `create` method of [`useNavigation`](/core/hooks/navigation/useNavigation.md) and then redirect to `/posts/create`.

Reviewer: ```suggestion
Clicking the button will trigger the `edit` method of [`useNavigation`](/core/hooks/navigation/useNavigation.md) and then redirect to `/posts/edit/2`.
```

| props                       | Material UI button props                         | [`ButtonProps`](https://mui.com/material-ui/api/button/)          |
| resourceNameOrRouteName     | Determines which resource to use for redirection | `string`                                                          | Resource name that it reads from route                                                                                |
| hideText                    | Allows to hide button text                       | `boolean`                                                         | `false`                                                                                                               |
| ignoreAccessControlProvider | Skip access control                              | `boolean`                                                         | `false`                                                                                                               |

> | children                    | Sets the button text                             | `ReactNode`                                                       | `"Create"`                                                                                                            |

Reviewer: ```suggestion
| children                    | Sets the button text                             | `ReactNode`                                                       | `"Edit"`                                                                                                            |
```
